tensorly/decomposition/candecomp_parafac.py

```diff
@@ -160,13 +163,17 @@ def parafac(tensor, rank, n_iter_max=100, init='svd', svd='numpy_svd',\
     weights = tl.ones(rank, **tl.context(tensor))
     Id = tl.eye(rank)*l2_reg
 
+    # MGR: fixed Modes
+    # Generating the mode update sequence
+    gen = (mode for mode in range(tl.ndim(tensor)) if mode not in fixed_modes)
```

Comment:
Why a generator rather than a list?

Reply:
I actually used lists in the other PR (which was more advanced). It does not really matter as long as we are not interested in picking an element in the generator. I can change that for sure.